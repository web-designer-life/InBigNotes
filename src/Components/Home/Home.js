import React from 'react';
import { ListItem } from '../Note/ListItem';
import noteDB from '../../notes.json';

export default class Home extends React.Component {
    render() {
        return (
            <div>
                {
                    noteDB ? 
                    <ListItem 
                        itemList={noteDB}
                    />
                    : <p>No notes</p>
                }
            </div>
        )
    }
}
