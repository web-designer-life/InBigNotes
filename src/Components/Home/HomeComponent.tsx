import * as React from 'react';
import { ListNotes } from '../Notes/ListNotes';
import noteDB from '../../notes.json';

export default class HomeComponent extends React.Component {
    render() {
        return (
            <div>
                {
                    noteDB ? 
                    <ListNotes 
                        notes={noteDB}
                    />
                    : <p>No notes</p>
                }
            </div>
        )
    }
}
