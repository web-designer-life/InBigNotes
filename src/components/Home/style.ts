import styled from 'styled-components';
import { COLORS } from '../../constants';

const EmptyListNotesWrapper = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    flex-grow: 1;
`;

const EmptyListNotesText = styled.h2`
    font-size: 40px;
    font-weight: 400;
    color: ${COLORS.STEEL_GRAY};
`;

export { EmptyListNotesWrapper, EmptyListNotesText };
