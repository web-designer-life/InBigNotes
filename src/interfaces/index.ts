export interface INote {
    id: string;
    title: string;
    text: string;
    created_at: number;
    updated_at: number;
};
