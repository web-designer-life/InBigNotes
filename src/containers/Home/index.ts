import Home from './Home';
import { connect } from 'react-redux';

interface Note {
    id: number, 
    title: string, 
    text: string,
    created_at: string,
    updated_at: string
}

interface State {
    notes: Array<Note>
}

const mapStateToProps = (state: State) => ({notes: state.notes});

export default connect(mapStateToProps)(Home);