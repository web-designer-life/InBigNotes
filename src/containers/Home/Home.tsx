import * as React from 'react';
import HomeComponent from '../../Components/Home/HomeComponent';

export default class Home extends React.Component {
    render() {
        return (
            <>
                <HomeComponent 
                    // notes={this.props.notes}
                />
            </>
        )
    }
}
