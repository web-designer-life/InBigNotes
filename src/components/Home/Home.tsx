import * as React from 'react';
import ControlPanel from '../ControlPanel/ControlPanel';
import { ListNotes } from '../Notes/ListNotes';
import { 
    RouterWrapper, 
    Logo, 
    Wrapper,
} from '../../containers/style';
import Button from '../Button/Button';
import { Note } from '../../interfaces';
import { Link } from 'react-router-dom';
import { EmptyListNotesWrapper, EmptyListNotesText } from './style';
import { 
    ROUTES, 
    BUTTON_TYPES, 
    COLORS, 
    TEXTS, 
} from '../../constants';
import Modal from '../Modal/Modal';

interface Props {
    notes?: Note[],
    selectedNotes?: string[],
    filter: string,
    fetchNotes(filter: string): Function,
    deleteNotes(notes: Note[]): Function,
    addSelectedNote(note: Note): Function,
    removeUnselectedNote(note: Note): Function,
    selectAllNotes(notes: Note[]): Function,
    unselectAllNotes(): Function,
    filterAction(filter: string): Function,
};

interface State {
    modalText: string,
    buttonConfirmText: string,
    buttonCancelText: string,
    active: boolean,
    action: () => void,
}

export default class Home extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);

        this.state = {
            modalText: '',
            buttonConfirmText: '',
            buttonCancelText: '',
            active: false,
            action: () => {},
        };

        this.handleModalBackButtonClick = this.handleModalBackButtonClick.bind(this);
        this.handleDeleteNotes = this.handleDeleteNotes.bind(this);
        this.toggleModal = this.toggleModal.bind(this);
    };    

    toggleModal() {
		this.setState({
			active: !this.state.active,
		} as Pick<State, keyof State>);
	};

    handleSetAction(func: Function) {
		this.setState({
			action: func,
		} as Pick<State, keyof State>);
	};

    handleModalBackButtonClick() {
        this.setState({
			modalText: TEXTS.MODAL.DELETE,
            buttonConfirmText: TEXTS.BUTTON.DELETE,
            buttonCancelText: TEXTS.BUTTON.CANCEL,
		} as Pick<State, keyof State>);

        this.handleSetAction(this.handleDeleteNotes);
        this.toggleModal();
    }

    handleDeleteNotes() {
        const {
            notes, 
            selectedNotes, 
            unselectAllNotes, 
            deleteNotes,
        } = this.props;

        const completeNotes = notes?.filter(
            (note: Note) => !selectedNotes?.some(
                (selectedNote: String) => note.id === selectedNote
            )
        ) || [];

        unselectAllNotes();
        deleteNotes(completeNotes);
    };
    
    render() {
        const { 
            notes, 
            selectedNotes,
            filter,
            fetchNotes,
            addSelectedNote, 
            removeUnselectedNote,
            selectAllNotes,
            unselectAllNotes,
            filterAction,
        } = this.props;

        const {
            modalText, 
            buttonConfirmText,
            buttonCancelText,
            active, 
            action,
        } = this.state;

        return (
            (notes && notes.length !== 0) ? 
            <>
                <RouterWrapper>
                    <Logo>{TEXTS.PROJECT_NAME}</Logo>
                    <Wrapper>
                        <Link to={ROUTES.NOTE_CREATE}>
                            <Button 
                                type={BUTTON_TYPES.BUTTON}
                                disabled={!!selectedNotes?.length}
                                text={TEXTS.BUTTON.CREATE}
                                color={COLORS.BUTTON.GREEN}
                            />
                        </Link>
                        <Button 
                            type={BUTTON_TYPES.BUTTON}
                            disabled={!selectedNotes?.length}
                            onClick={this.handleModalBackButtonClick}
                            text={TEXTS.BUTTON.DELETE}
                            color={COLORS.BUTTON.RED}
                        />
                    </Wrapper>
                </RouterWrapper>
                <ControlPanel 
                    notes={notes}
                    selectedNotes={selectedNotes}
                    filter={filter}
                    fetchNotes={fetchNotes}
                    selectAllNotes={selectAllNotes}
                    unselectAllNotes={unselectAllNotes}
                    filterAction={filterAction}
                />
                <ListNotes 
                    notes={notes}
                    selectedNotes={selectedNotes}
                    addSelectedNote={addSelectedNote}
                    removeUnselectedNote={removeUnselectedNote}
                />
                <Modal 
                    modalText={modalText} 
                    buttonConfirmText={buttonConfirmText} 
                    buttonCancelText={buttonCancelText}
                    active={active} 
                    onClose={this.toggleModal} 
                    action={action} 
                />
            </>
            :
            <>
                <RouterWrapper>
                    <Logo>{TEXTS.PROJECT_NAME}</Logo>
                    <Wrapper>
                        <Link to={ROUTES.NOTE_CREATE}>
                            <Button 
                                type={BUTTON_TYPES.BUTTON}
                                disabled={!!selectedNotes?.length}
                                text={TEXTS.BUTTON.CREATE}
                                color={COLORS.BUTTON.GREEN}
                            />
                        </Link>
                        <Button
                            type={BUTTON_TYPES.BUTTON}
                            disabled={!selectedNotes?.length}
                            onClick={this.handleModalBackButtonClick}
                            text={TEXTS.BUTTON.DELETE}
                            color={COLORS.BUTTON.RED}
                        />
                    </Wrapper>
                </RouterWrapper>
                <EmptyListNotesWrapper>
                    <EmptyListNotesText>{TEXTS.EMPTY_STORAGE_DATA}</EmptyListNotesText>
                </EmptyListNotesWrapper>
                <Modal 
                    modalText={modalText} 
                    buttonConfirmText={buttonConfirmText} 
                    buttonCancelText={buttonCancelText}
                    active={active} 
                    onClose={this.toggleModal} 
                    action={action} 
                />
            </>
        )
    }
};
