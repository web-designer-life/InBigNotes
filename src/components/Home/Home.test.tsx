import React from 'react'
import Home from './Home';
import { shallow } from 'enzyme';
import { Note } from '../../interfaces';

const homeProps = {
    notes: [
        {
            id: 'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
            title: 'Army',
            text: 'An army, ground force, or land force is a fighting force that fights primarily on land.',
            created_at: 1632734670806,
            updated_at: 1632735522467
        },
    ],
    selectedNotes: [
        'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
    ],
    filter: 'title',
    fetchNotes: (filter: string) => Function,
    deleteNotes: (notes: Note[]) => Function,
    addSelectedNote: (note: Note) => Function,
    removeUnselectedNote: (note: Note) => Function,
    selectAllNotes: (notes: Note[]) => Function,
    unselectAllNotes: () => Function,
    filterAction: (filter: string) => Function,
    navigateToPage: (path: string) => Function,
};

describe("Home component", () => {
    it("should render Home component", () => {
        const component = shallow(<Home {...homeProps} />);
        expect(component).toMatchSnapshot();
    });
});
