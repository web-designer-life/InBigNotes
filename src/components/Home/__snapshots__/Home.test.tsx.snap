// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render Home component 1`] = `
<Fragment>
  <styled.div>
    <styled.h1>
      InBigNotes
    </styled.h1>
    <styled.div>
      <Button
        color="green"
        disabled={true}
        onClick={[Function]}
        text="Create"
        type="button"
      />
      <Button
        color="red"
        disabled={false}
        onClick={[Function]}
        text="Delete"
        type="button"
      />
    </styled.div>
  </styled.div>
  <ControlPanel
    fetchNotes={[Function]}
    filter="title"
    filterAction={[Function]}
    notes={
      Array [
        Object {
          "created_at": 1632734670806,
          "id": "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
          "text": "An army, ground force, or land force is a fighting force that fights primarily on land.",
          "title": "Army",
          "updated_at": 1632735522467,
        },
      ]
    }
    selectAllNotes={[Function]}
    selectedNotes={
      Array [
        "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
      ]
    }
    unselectAllNotes={[Function]}
  />
  <ListNotes
    addSelectedNote={[Function]}
    navigateToPage={[Function]}
    notes={
      Array [
        Object {
          "created_at": 1632734670806,
          "id": "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
          "text": "An army, ground force, or land force is a fighting force that fights primarily on land.",
          "title": "Army",
          "updated_at": 1632735522467,
        },
      ]
    }
    removeUnselectedNote={[Function]}
    selectedNotes={
      Array [
        "b83cbac2-d74e-458d-85c0-75bdf9c29ba3",
      ]
    }
  />
  <Modal
    action={[Function]}
    active={false}
    buttonCancelText=""
    buttonConfirmText=""
    modalText=""
    onClose={[Function]}
  />
</Fragment>
`;
